### ISSUE TITLE
pocl_pthread_update_event() fails to broadcast to all listeners, so only one thread gets woken up

### ISSUE THREAD

**Issue by sdh4** [pocl/pocl#685]
pocl_pthread_update_event() in lib/CL/devices/pthread/pthread.c wakes up threads waiting on the event with pthread_cond_signal(), which only wakes a single thread. If more than one thread is waiting on an event, the others never wake and wait forever. 

The fix is simple: Change the calls to pthread_cond_signal() to pthread_cond_broadcast().

**Comment by sdh4** [comment]
Added pull request #686

**Comment by franz** [comment]
merged into master
